lib/Runtime/PlatformAgnostic/RuntimePlatformAgnosticPch.h

```diff
@@ -15,6 +17,42 @@
 #pragma warning(pop)
 #endif
 
+// Minimal definitions to use AssertMsg in the PAIL
+#ifndef _WIN32
+#define DbgRaiseAssertionFailure() __builtin_trap()
+#define __analysis_assume(x)
+#define __in
+#define __inout
+
+#ifndef TRUE
```

Comment:
The Assertion code uses TRUE/FALSE today, as does parts of the ICU. I could remove the definitions and include <Unicode/umachine.h> instead to get TRUE/FALSE from there. If @obastemur changes the TRUE/FALSE in the Assertion code to using true/false, then we could move this #include out of the pch and into UnicodeText.ICU.cpp since that would be the only code that would need it.